packages/indexing/src/tables-manager.ts

```diff
@@ -325,41 +332,38 @@ export class SqliteTablesManager extends TablesManager {
   /**
    * Create mid tables and corresponding indexes
    */
-  override async initMidTables(modelsToIndex: Array<IndexModelArgs>) {
+  override async initMidTables(
+    modelsToIndex: Array<IndexModelArgs>,
+    createIndices: CreateIndicesRecord
+  ) {
     const existingTables = await this.listMidTables()
     await Promise.all(
       modelsToIndex.map(async (modelIndexArgs) => {
-        await this.initMidTable(modelIndexArgs, existingTables)
+        await this.initMidTable(
+          modelIndexArgs,
+          existingTables,
+          createIndices[modelIndexArgs.model.toString()]
+        )
       })
     )
   }
 
   /**
    * Create a single mid table for a given model
    */
-  async initMidTable(modelIndexArgs: IndexModelArgs, existingTables: Array<string>) {
+  async initMidTable(
+    modelIndexArgs: IndexModelArgs,
+    existingTables: Array<string>,
+    createIndices: Array<FieldsIndex> = []
+  ) {
     const tableName = asTableName(modelIndexArgs.model)
-    const relationColumns = relationsDefinitionsToColumnInfo(modelIndexArgs.relations)
-    if (existingTables.includes(tableName)) {
-      if (relationColumns.length) {
-        // Make relations columns nullable
-        await this.dataSource.schema.alterTable(tableName, (table) => {
-          for (const column of relationColumns) {
-            if (column.type === ColumnType.STRING) {
-              const columnName = addColumnPrefix(column.name)
-              table.string(columnName, 1024).nullable().alter()
-            }
-          }
-        })
-      }
-      return
-    }
 
-    this.logger.imp(`Creating ComposeDB Indexing table for model: ${tableName}`)
-    await createSqliteModelTable(this.dataSource, tableName, relationColumns)
-    if (modelIndexArgs.indices) {
-      await createSqliteIndices(this.dataSource, tableName, modelIndexArgs.indices)
+    if (!existingTables.includes(tableName)) {
+      this.logger.imp(`Creating ComposeDB Indexing table for model: ${tableName}`)
+      await createSqliteModelTable(this.dataSource, tableName)
     }
+
+    await createSqliteIndices(this.dataSource, tableName, createIndices)
```

Comment:
Given there's a promise involved here, it would be better to have this be only called if there's indices to create.